src/azure-cli-core/azure/cli/core/keys.py

```diff
@@ -31,18 +35,46 @@ def is_valid_ssh_rsa_public_key(openssh_pubkey):
 
 def generate_ssh_keys(private_key_filepath, public_key_filepath):
     import paramiko
+    from paramiko.ssh_exception import PasswordRequiredException, SSHException
+
+    if os.path.isfile(public_key_filepath):
```

Comment:
We probably don't need this as the caller is supposed to check the public key file not existing and then invoke this. Check out the code [here](https://github.com/Azure/azure-cli/blob/dev/src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_validators.py#L33)

Reply:
I am fine with this idea, but then you also need to remove the redundant file check in VM and ACS modules. I thought we can do step by step through the next PR, but up to you.